src/data/datasets/classification/image_classification.py

```diff
@@ -0,0 +1,132 @@
+from typing import Union, Optional
+from functools import partial
+
+import re
+import torch
+import numpy as np
+from albumentations import BasicTransform
+from albumentations.core.composition import BaseCompose
+
+from src.registry import DATASETS
+from src.data.datasets.base import ImageDataset
+
+
+@DATASETS.register_class
+class ImageClassificationDataset(ImageDataset):
+    """A generic dataset for image classification task"""
+
+    def __init__(self,
+                 data_folder: str,
+                 path_to_datalist: str,
+                 transform: Optional[Union[BasicTransform, BaseCompose]],
+                 augment: Optional[Union[BasicTransform, BaseCompose]] = None,
+                 input_dtype: str = 'float32',
+                 target_dtype: str = 'long',
+                 target_column: str = 'label',
+                 grayscale: bool = False,
+                 expand_rate: int = 1,
+                 test_mode: Optional[bool] = False,
+                 multilabel: bool = False,
+                 num_classes: int = None,
+                 lazy_init_multilabel: bool = False):
+        """
+        Args:
+            data_folder: Directory with all the images.
+            path_to_datalist: Path to the csv file with path to images and annotations.
+                Path to images must be under column `image_path` and annotations must be under `label` column
+            transform: Transform to be applied on a sample. This should have the
+                interface of transforms in `albumentations` library.
+            augment: Optional augment to be applied on a sample.
+                This should have the interface of transforms in `albumentations` library.
+            input_dtype: Data type of of the torch tensors related to the image.
+            target_dtype: Data type of of the torch tensors related to the target.
+            target_column: Name of the column that contains target labels.
+            grayscale: If True, image will be read as grayscale otherwise as RGB.
+            expand_rate: A multiplier that shows how many times the dataset will be larger than its real size.
+                Useful for small datasets.
+            test_mode: If True, only image without labels will be returned.
+            multilabel: If True, target labels is being converted to multihot vector for multilabel task.
```

Comment:
`If True, target labels are being converted to multihot vector for multilabel task.`